src/agreementManager/AgreementManager.ts

```diff
@@ -3,237 +3,108 @@ import {
         block: Block,
         participant: Address
     ): { didSign: boolean; signature: Signature | undefined } {
-        const agreement = this.forks.agreementByBlock(block);
-
-        if (!agreement || !agreement.block.equals(block))
-            return { didSign: false, signature: undefined };
+        const blockEntry = this.storage.blocks.getBlockEntry(block.hash);
+        if (!blockEntry) return { didSign: false, signature: undefined };
+
+        // Check if participant is the author
+        if (block.author === participant) {
+            return {
+                didSign: true,
+                signature: this.storage.blocks.getOriginalSignature(block)
+            };
+        }
 
-        return SignatureService.getParticipantSignature(agreement, participant);
+        // Check all signatures
+        return block.findSignature(
+            participant,
+            this.getAllSignatures(blockEntry.blockConfirmation)
+        );
     }
 
-    public getParticipantsWhoHaventSignedBlock(block: Block): Address[] {
-        const forkId = block.forkId;
-        const agreement = this.forks.agreementByBlock(block);
-        const fork = this.forks.forkAt(forkId);
-        if (!fork || !agreement) return [];
+    /**
+     * Get participants who haven't signed a block
+     */
+    public getParticipantsWhoDidntSign(block: Block): Address[] {
+        const blockEntry = this.storage.blocks.getBlockEntry(block.hash);
+        if (!blockEntry) return [];
 
-        return SignatureService.getParticipantsWhoDidntSign(fork, agreement);
-    }
+        const thresholdAddresses = this.storage.getParticipants(
+            block.coordinates
+        );
 
-    public isParticipantInLatestFork(participant: Address): boolean {
-        const fork = this.forks.latestFork();
-        if (!fork) return false;
-        return new Set(fork.addressesInThreshold).has(participant);
-    }
+        const signersSet = block.getSignerAddresses(
+            this.getAllSignatures(blockEntry.blockConfirmation)
+        );
 
-    public getEncodedState(
-        forkId: ForkId,
-        transactionCnt: BlockHeight
-    ): Bytes | undefined {
-        const agreement = this.forks.agreement(forkId, transactionCnt);
-        return agreement?.encodedState;
-    }
-    public getForkGenesisStateEncoded(forkId: ForkId): Bytes | undefined {
-        const fork = this.forks.forkAt(forkId);
-        return fork?.forkGenesisStateEncoded;
+        // Return addresses that haven't signed
+        return thresholdAddresses.filter((address) => !signersSet.has(address));
     }
+
     /**
-     * Gets the latest finalized state (ecnoded) and the latest signed/confirmed state (encoded) from the signer with virtual votes proving it
-     * @param forkId
-     * @param signerAddress
-     * @returns
+     * Get the latest finalized state and latest signed state (by signer) with virtual voting blocks
      */
     public getFinalizedAndLatestWithVotes(
```

Comment:
This has to be changed for milestone logic - can't be a direct 1:1 mapping from our old logic -> fine for now so Adrian is not blocked - I'll crate a Trello card which I'll populate later